### PR TITLE
ProxyReader contract

### PR DESCRIPTION
**Pull request by aquiladev**

### CODE REVIEW

File: contracts/ProxyReader.sol [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        Resolver resolver = Resolver(_registry.resolverOf(tokenId));
+        return resolver.getByHash(keyHash, tokenId);
+    }
+
+    function getManyByHash(uint256[] memory keyHashes, uint256 tokenId) public view returns (string[] memory keys, string[] memory values) {
+        Resolver resolver = Resolver(_registry.resolverOf(tokenId));
+        return resolver.getManyByHash(keyHashes, tokenId);
+    }
+
+    // TODO: figure out right way to do this without tokenId
```
Comment (bogdan) [inline contracts/ProxyReader.sol:141]: Cleanup?

File: contracts/ProxyReader.sol [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    function root() public view returns (uint256) {
+        return _registry.root();
+    }    
+
+    function getMany(string[] calldata keys, uint256 tokenId) external view returns (string[] memory) {
+        Resolver resolver = Resolver(_registry.resolverOf(tokenId));
+        return resolver.getMany(keys, tokenId);
+    }
+
+    // TODO: (string[] calldata keys, uint256 tokenId) -> resolver address, owner address, values[]
```
Comment (bogdan) [inline contracts/ProxyReader.sol:119]: Lets implement this one.

Comment (aquiladev) [reply contracts/ProxyReader.sol:119]: lets come up with a name for the func

File: DESIGN_FLAWS.md [hunk 3]
```diff
@@ -107,6 +133,40 @@ We need a method like:
...
+     * bytes4(keccak256(abi.encodePacked('registry()'))) == 0x7b103999
+     * bytes4(keccak256(abi.encodePacked('get(string,uint256)'))) == 0x1be5e7ed
+     * bytes4(keccak256(abi.encodePacked('getByHash(uint256,uint256)'))) == 0x672b9f81
+     * bytes4(keccak256(abi.encodePacked('getMany(string[],uint256)'))) == 0x1bd8cc1a
+     * bytes4(keccak256(abi.encodePacked('getManyByHash(uint256[],uint256)'))) == 0xb85afd28
+     *
+     * => 0x6ccbae5f ^ 0x7b103999 ^ 0x1be5e7ed ^
+     *    0x672b9f81 ^ 0x1bd8cc1a ^ 0xb85afd28 == 0xc897de98
+     */
+    bytes4 private constant _INTERFACE_ID_RESOLVER_READER = 0xc897de98;
```
Comment (mvlabat) [inline DESIGN_FLAWS.md:159]: This is a great idea

File: DESIGN_FLAWS.md [hunk 4]
```diff
@@ -107,6 +133,40 @@ We need a method like:
 setManyToEachFor(tokenIds, keys, values, signature)
 ```
 
+#### Ability to check interface support
+
+Resolver should implement [ERC165](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-165.md) to have the ability to check interface support.
+Resolver must implement IResolverReader interface in order to be fully supported by ProxyReader.
```
Comment (mvlabat) [inline DESIGN_FLAWS.md:139]: 👍 

What are our plans on updating our deployed resolvers, btw? If we introduce ERC165 support, users might expect that every UD resolver implements it

Comment (bogdan) [reply DESIGN_FLAWS.md:139]: I don't think we need to discuss these things in this PR. We will have an opportunity to review the whole "Design Flaws" document and make sure we implement it right. For now: we are only insterested in good ideas list.

**Pull request merged**